tests/Core/Ruleset/ProcessRulesetShouldProcessElementTest.php

```diff
@@ -0,0 +1,388 @@
+        $this->assertTrue(self::$config->colors, 'CS-only arg directive was not applied.');
+        $this->assertSame($expectedReports, self::$config->reports, 'CBF-only arg directive was applied, while it shouldn\'t have been.');
+
+    }//end testShouldProcessArgCsonly()
+
+
+    /**
+     * Verify that in CBF mode, phpcbf-only <arg> directives are respected and phpcs-only <arg>
+     * directives are ignored.
+     *
+     * @group CBF
+     *
+     * @return void
+     */
+    public function testShouldProcessArgCbfonly()
+    {
+        if (PHP_CODESNIFFER_CBF === false) {
+            $this->markTestSkipped('This test needs CBF mode to run');
+        }
+
+        $expectedExtensions = [
+            'php'  => 'PHP',
+            'phpt' => 'PHP',
+        ];
+        $expectedReports    = ['summary' => null];
+
+        $this->assertSame($expectedExtensions, self::$config->extensions, 'Non-selective arg directive was not applied.');
+        $this->assertTrue(self::$config->showProgress, 'Non-selective short arg directive was not applied [1].');
+        $this->assertTrue(self::$config->showSources, 'Non-selective short arg directive was not applied [2].');
+        $this->assertFalse(self::$config->colors, 'CS-only arg directive was applied, while it shouldn\'t have been.');
+        $this->assertSame($expectedReports, self::$config->reports, 'CBF-only arg directive was not applied.');
+
+    }//end testShouldProcessArgCbfonly()
+
+
+    /**
+     * Verify that in CS mode, phpcs-only <ini> directives are respected and phpcbf-only <ini>
+     * directives are ignored.
+     *
+     * @return void
+     */
+    public function testShouldProcessIniCsonly()
+    {
+        if (PHP_CODESNIFFER_CBF === true) {
+            $this->markTestSkipped('This test needs CS mode to run');
+        }
+
+        $this->assertSame('2', ini_get('bcmath.scale'), 'Non-selective ini directive was not applied.');
+        $this->assertSame('path/to/docs/', ini_get('docref_root'), 'CS-only ini directive was not applied.');
+        $this->assertSame('', ini_get('user_agent'), 'CBF-only ini directive was applied, while it shouldn\'t have been.');
+
+    }//end testShouldProcessIniCsonly()
+
+
+    /**
+     * Verify that in CBF mode, phpcbf-only <ini> directives are respected and phpcs-only <ini>
+     * directives are ignored.
+     *
+     * @group CBF
+     *
+     * @return void
+     */
+    public function testShouldProcessIniCbfonly()
+    {
+        if (PHP_CODESNIFFER_CBF === false) {
+            $this->markTestSkipped('This test needs CBF mode to run');
+        }
+
+        $this->assertSame('2', ini_get('bcmath.scale'), 'Non-selective ini directive was not applied.');
+        $this->assertSame('', ini_get('docref_root'), 'CS-only ini directive was not applied.');
+        $this->assertSame('Never mind', ini_get('user_agent'), 'CBF-only ini directive was applied, while it shouldn\'t have been.');
```

Comment:
The messages here are incorrect. In the case of this test, CS-only is the one that should not be applied, and the CBF-only is the one that should be applied.

Reply:
Ouch.. thanks for catching that. Double the others too, couldn't find any other wrong ones, should be fine now.